kythe/cxx/common/kzip_writer_c_api_test.cc

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2018 The Kythe Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "kythe/cxx/common/kzip_writer_c_api.h"
+
+#include <zip.h>
+
+#include <cstdlib>
+#include <unordered_map>
+#include <unordered_set>
+
+#include "absl/status/status.h"
+#include "absl/strings/str_cat.h"
+#include "absl/strings/str_replace.h"
+#include "absl/strings/string_view.h"
+#include "absl/strings/strip.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "kythe/cxx/common/kzip_reader.h"
+#include "kythe/cxx/common/libzip/error.h"
+#include "kythe/proto/go.pb.h"
+
+namespace kythe {
+namespace {
+using ::testing::ElementsAre;
+using ::testing::Values;
+
+absl::string_view TestTmpdir() {
+  return absl::StripSuffix(std::getenv("TEST_TMPDIR"), "/");
+}
```

Comment:
This should just be `::testing::TempDir()`

Reply:
It should be, but it won't work as is.

It seems that the [default gtest behavior][dtb] is not modified in kythe, so we'll always get `/tmp` as the temp dir. In bazel we want to get the sharded temp dir from`TEST_TMPDIR`; but this doesn't seem to be the default on Linux.

[dtb]: https://github.com/google/googletest/blob/master/googletest/src/gtest.cc#L6561